test/app-luatest/debug_test.lua

```diff
@@ -0,0 +1,76 @@
+local t = require('luatest')
+
+local justrun = require('luatest.justrun')
+local server = require('luatest.server')
+local treegen = require('luatest.treegen')
+
+local tests = {
+    { chunk = [[print(debug.sourcefile())]] },
+    { chunk = [[print(debug.sourcedir())]] },
+    { chunk = [[print(debug.__file__)]] },
+    { chunk = [[print(debug.__dir__)]] },
+    { chunk = [[print(require('net.box').self:call('debug.sourcefile'))]] },
+    { chunk = [[print(require('net.box').self:call('debug.sourcedir'))]] },
+    { chunk = [[fn = function() return debug.__file__ end;
+                print(require('net.box').self:call('fn'))]] },
+    { chunk = [[fn = function() return debug.__dir__ end;
+               print(require('net.box').self:call('fn'))]] },
+    { chunk = [[fn = function() local res = debug.sourcefile(); return res end;
+                print(require('net.box').self:call('fn'))]] },
+    { chunk = [[fn = function() local res = debug.sourcedir(); return res end;
+                print(require('net.box').self:call('fn'))]] },
+    { chunk = [[print(loadstring('return debug.sourcefile()')())]] },
+    { chunk = [[print(loadstring('return debug.sourcedir()')())]] },
+    { chunk = [[print(loadstring('return debug.__file__')())]] },
+    { chunk = [[print(loadstring('return debug.__dir__')())]] },
+    { chunk = [[print(loadstring('local res = debug.sourcefile(); return res')())]] },
+    { chunk = [[print(loadstring('local res = debug.sourcedir(); return res')())]] },
+}
+
+local g_remote = t.group('remote')
+
+g_remote.before_all(function()
+    g_remote.server = server:new({alias = 'remote_debug'})
+    g_remote.server:start()
+end)
+
+g_remote.after_all(function()
+    g_remote.server:stop()
+end)
+
+-- When running netbox call on remote server.
+g_remote.test_debug_module_via_net_box = function()
+    local NULL = require('json').NULL
+    local conn = g_remote.server.net_box
+
+    local result, err = conn:call('debug.sourcefile')
+    t.assert_equals(result, NULL, 'debug.sourcefile() returns NULL')
+    t.assert_equals(err, nil, 'debug.sourcefile() returns no error')
+
+    local result, err = conn:call('debug.sourcedir')
+    t.assert_equals(result, '.', 'debug.sourcedir() returns "."')
+    t.assert_equals(err, nil, 'debug.sourcedir() returns no error')
+end
+
+local g = t.group('local', tests)
+
+-- When running lua code from console.
+-- Note, `debug.sourcefile()` returns cwd when running within
+-- console.
+g.test_debug_module_via_console = function(cg)
+    local TARANTOOL_PATH = arg[-1]
+    local cmd = string.format('%s -e "%s; os.exit(0)"',
+                              TARANTOOL_PATH,
+                              cg.params.chunk)
+    t.assert_equals(os.execute(cmd), 0)
```

Comment:
Don't get it, could clarify?

Reply:
Now the command are in testcase names:

```
Tarantool version is 3.3.0-entrypoint-188-gf3ecdb1b40
Started on Fri Nov  1 19:36:41 2024
    local.chunk:"fn = function() local res = debug.sourcedir(); return res end;\
                print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_console ... (0.085s) Ok
    local.chunk:"fn = function() local res = debug.sourcedir(); return res end;\
                print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_script ... (0.086s) Ok
    local.chunk:"fn = function() local res = debug.sourcefile(); return res end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_console ... (0.082s) Ok
    local.chunk:"fn = function() local res = debug.sourcefile(); return res end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_script ... (0.085s) Ok
    local.chunk:"fn = function() return debug.__dir__ end;\
               print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_console ... (0.084s) Ok
    local.chunk:"fn = function() return debug.__dir__ end;\
               print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"fn = function() return debug.__file__ end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_console ... (0.084s) Ok
    local.chunk:"fn = function() return debug.__file__ end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_script ... (0.085s) Ok
    local.chunk:"print(debug.__dir__)".res:".".test_debug_module_via_console ... (0.088s) Ok
    local.chunk:"print(debug.__dir__)".res:".".test_debug_module_via_script ... (0.083s) Ok
    local.chunk:"print(debug.__file__)".res:"nil".test_debug_module_via_console ... (0.081s) Ok
    local.chunk:"print(debug.__file__)".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(debug.sourcedir())".res:".".test_debug_module_via_console ... (0.083s) Ok
    local.chunk:"print(debug.sourcedir())".res:".".test_debug_module_via_script ... (0.085s) Ok
    local.chunk:"print(debug.sourcefile())".res:"nil".test_debug_module_via_console ... (0.083s) Ok
    local.chunk:"print(debug.sourcefile())".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(loadstring('local res = debug.sourcedir(); return res')())".res:".".test_debug_module_via_console ... (0.085s) Ok
    local.chunk:"print(loadstring('local res = debug.sourcedir(); return res')())".res:".".test_debug_module_via_script ... (0.085s) Ok
    local.chunk:"print(loadstring('local res = debug.sourcefile(); return res')())".res:"nil".test_debug_module_via_console ... (0.085s) Ok
    local.chunk:"print(loadstring('local res = debug.sourcefile(); return res')())".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(loadstring('return debug.__dir__')())".res:".".test_debug_module_via_console ... (0.088s) Ok
    local.chunk:"print(loadstring('return debug.__dir__')())".res:".".test_debug_module_via_script ... (0.087s) Ok
    local.chunk:"print(loadstring('return debug.__file__')())".res:"nil".test_debug_module_via_console ... (0.082s) Ok
    local.chunk:"print(loadstring('return debug.__file__')())".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(loadstring('return debug.sourcedir()')())".res:".".test_debug_module_via_console ... (0.082s) Ok
    local.chunk:"print(loadstring('return debug.sourcedir()')())".res:".".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(loadstring('return debug.sourcefile()')())".res:"nil".test_debug_module_via_console ... (0.084s) Ok
    local.chunk:"print(loadstring('return debug.sourcefile()')())".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(require('net.box').self:call('debug.sourcedir'))".res:".".test_debug_module_via_console ... (0.082s) Ok
    local.chunk:"print(require('net.box').self:call('debug.sourcedir'))".res:".".test_debug_module_via_script ... (0.084s) Ok
    local.chunk:"print(require('net.box').self:call('debug.sourcefile'))".res:"".test_debug_module_via_console ... (0.082s) Ok
    local.chunk:"print(require('net.box').self:call('debug.sourcefile'))".res:"".test_debug_module_via_script ... (0.085s) Ok
    remote.test_debug_module_via_net_box ... (0.001s) Ok
=========================================================
Ran 33 tests in 3.024 seconds, [1m[32m33 succeeded[0m, 0 failed
```

Reply:
```
Tarantool version is 3.3.0-entrypoint-188-gf3ecdb1b40
Started on Fri Nov  1 19:40:12 2024
    console.chunk:"fn = function() local res = debug.sourcedir(); return res end;\
                print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_console ... (0.085s) Ok
    console.chunk:"fn = function() local res = debug.sourcefile(); return res end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_console ... (0.082s) Ok
    console.chunk:"fn = function() return debug.__dir__ end;\
               print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_console ... (0.083s) Ok
    console.chunk:"fn = function() return debug.__file__ end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_console ... (0.083s) Ok
    console.chunk:"print(debug.__dir__)".res:".".test_debug_module_via_console ... (0.086s) Ok
    console.chunk:"print(debug.__file__)".res:"nil".test_debug_module_via_console ... (0.086s) Ok
    console.chunk:"print(debug.sourcedir())".res:".".test_debug_module_via_console ... (0.083s) Ok
    console.chunk:"print(debug.sourcefile())".res:"nil".test_debug_module_via_console ... (0.086s) Ok
    console.chunk:"print(loadstring('local res = debug.sourcedir(); return res')())".res:".".test_debug_module_via_console ... (0.083s) Ok
    console.chunk:"print(loadstring('local res = debug.sourcefile(); return res')())".res:"nil".test_debug_module_via_console ... (0.084s) Ok
    console.chunk:"print(loadstring('return debug.__dir__')())".res:".".test_debug_module_via_console ... (0.082s) Ok
    console.chunk:"print(loadstring('return debug.__file__')())".res:"nil".test_debug_module_via_console ... (0.083s) Ok
    console.chunk:"print(loadstring('return debug.sourcedir()')())".res:".".test_debug_module_via_console ... (0.082s) Ok
    console.chunk:"print(loadstring('return debug.sourcefile()')())".res:"nil".test_debug_module_via_console ... (0.082s) Ok
    console.chunk:"print(require('net.box').self:call('debug.sourcedir'))".res:".".test_debug_module_via_console ... (0.083s) Ok
    console.chunk:"print(require('net.box').self:call('debug.sourcefile'))".res:"".test_debug_module_via_console ... (0.083s) Ok
    remote.test_debug_module_via_net_box ... (0.001s) Ok
    script.chunk:"fn = function() local res = debug.sourcedir(); return res end;\
                print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_script ... (0.086s) Ok
    script.chunk:"fn = function() local res = debug.sourcefile(); return res end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_script ... (0.091s) Ok
    script.chunk:"fn = function() return debug.__dir__ end;\
               print(require('net.box').self:call('fn'))".res:".".test_debug_module_via_script ... (0.084s) Ok
    script.chunk:"fn = function() return debug.__file__ end;\
                print(require('net.box').self:call('fn'))".res:"".test_debug_module_via_script ... (0.086s) Ok
    script.chunk:"print(debug.__dir__)".res:".".test_debug_module_via_script ... (0.085s) Ok
    script.chunk:"print(debug.__file__)".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    script.chunk:"print(debug.sourcedir())".res:".".test_debug_module_via_script ... (0.084s) Ok
    script.chunk:"print(debug.sourcefile())".res:"nil".test_debug_module_via_script ... (0.087s) Ok
    script.chunk:"print(loadstring('local res = debug.sourcedir(); return res')())".res:".".test_debug_module_via_script ... (0.085s) Ok
    script.chunk:"print(loadstring('local res = debug.sourcefile(); return res')())".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    script.chunk:"print(loadstring('return debug.__dir__')())".res:".".test_debug_module_via_script ... (0.085s) Ok
    script.chunk:"print(loadstring('return debug.__file__')())".res:"nil".test_debug_module_via_script ... (0.087s) Ok
    script.chunk:"print(loadstring('return debug.sourcedir()')())".res:".".test_debug_module_via_script ... (0.085s) Ok
    script.chunk:"print(loadstring('return debug.sourcefile()')())".res:"nil".test_debug_module_via_script ... (0.084s) Ok
    script.chunk:"print(require('net.box').self:call('debug.sourcedir'))".res:".".test_debug_module_via_script ... (0.084s) Ok
    script.chunk:"print(require('net.box').self:call('debug.sourcefile'))".res:"".test_debug_module_via_script ... (0.086s) Ok
=========================================================
Ran 33 tests in 3.050 seconds, [1m[32m33 succeeded[0m, 0 failed
```

Reply:
I've meant something like `t.assert_equals(os.execute(cmd), 0, ("cmd: '%s'):format(cmd)")` but since this information already persists in the luatest verbose mode, it is OK to leave your example as is (exept changing `"` to `[[` and `]]` to avoid misreading). Or just add the general message like `"check command status"`.

Reply:
Updated:

```diff
diff --git a/test/app-luatest/debug_test.lua b/test/app-luatest/debug_test.lua
index 79d8ee3cd0..48730164c7 100644
--- a/test/app-luatest/debug_test.lua
+++ b/test/app-luatest/debug_test.lua
@@ -130,8 +130,8 @@ g_console.test_debug_module_via_console = function(cg)
     local dir = treegen.prepare_directory({}, {})
     local opts = { nojson = true, quote_args = true }
     local res = justrun.tarantool(dir, {}, { '-e', cg.params.chunk }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, cg.params.res_console)
+    t.assert_equals(res.exit_code, 0, 'check command status')
+    t.assert_equals(res.stdout, cg.params.res_console, 'check command stdout')
 end
 
 local g_script = t.group('script', tests)
@@ -143,6 +143,6 @@ g_script.test_debug_module_via_script = function(cg)
     treegen.write_file(dir, script_name, cg.params.chunk)
     local opts = { nojson = true }
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, cg.params.res_script)
+    t.assert_equals(res.exit_code, 0, 'check command status')
+    t.assert_equals(res.stdout, cg.params.res_script, 'check command stdout')
 end
```